File: javascript-closure-angular-client/API/Client/InstallationObject.js
goog.provide('API.Client.installationObject');

/**
 * @record
 */
API.Client.InstallationObject = function() {}

/**
 * @type {!string}
 * @export
 */
API.Client.InstallationObject.prototype.id;

/**
 * @type {!string}
 * @export
 */
API.Client.InstallationObject.prototype.deviceToken;

/**
 * @type {!string}
 * @export
 */
API.Client.InstallationObject.prototype.deviceType;

/**
 * @type {!API.Client._metaInstallationObject}
 * @export
 */
API.Client.InstallationObject.prototype.meta;

/**
 * @type {!Array<!string>}
 * @export
 */
API.Client.InstallationObject.prototype.channels;

/** @enum {string} */
API.Client.InstallationObject.DeviceTypeEnum = { 
  ios: 'ios',
  android: 'android',
}

File: dart-client/lib/model/installation_object.dart
part of swagger.api;

class InstallationObject {
  
  String id = null;
  

  String deviceToken = null;
  

  String deviceType = null;
  //enum deviceTypeEnum {  ios,  android,  };

  MetaInstallationObject meta = null;
  

  List<String> channels = [];
  
  InstallationObject();

  @override
  String toString() {
    return 'InstallationObject[id=$id, deviceToken=$deviceToken, deviceType=$deviceType, meta=$meta, channels=$channels, ]';
  }

  InstallationObject.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    id =
        json['id']
    ;
    deviceToken =
        json['deviceToken']
    ;
    deviceType =
        json['deviceType']
    ;
    meta =
      
      
      new MetaInstallationObject.fromJson(json['meta'])
;
    channels =
        (json['channels'] as List).map((item) => item as String).toList()
    ;
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'deviceToken': deviceToken,
      'deviceType': deviceType,
      'meta': meta,
      'channels': channels
     };
  }

  static List<InstallationObject> listFromJson(List<dynamic> json) {
    return json == null ? new List<InstallationObject>() : json.map((value) => new InstallationObject.fromJson(value)).toList();
  }

  static Map<String, InstallationObject> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, InstallationObject>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new InstallationObject.fromJson(value));
    }
    return map;
  }
}

